File: dehaze-python/app/models/schema/message.py
from typing import Any

from pydantic import BaseModel, Field


class MessageSendRequest(BaseModel):
    templateCode: str | None = None
    type: str = Field(..., min_length=1, description="消息类型")
    title: str | None = None
    content: str | None = None
    recipientIds: list[int] = Field(..., min_length=1, description="接收人ID列表")
    bizModule: str | None = None
    bizId: str | None = None
    priority: int | None = Field(default=None, ge=1, le=4)
    jumpUrl: str | None = None
    variables: dict[str, str] | None = None
    extra: dict[str, Any] | None = None


class NotificationSettingsForm(BaseModel):
    pushEnabled: bool | None = None
    dndEnabled: bool | None = None
    dndStart: str | None = Field(
        default=None,
        pattern=r"^([01]\d|2[0-3]):[0-5]\d:[0-5]\d$",
        description="免打扰开始时间(HH:mm:ss)",
    )
    dndEnd: str | None = Field(
        default=None,
        pattern=r"^([01]\d|2[0-3]):[0-5]\d:[0-5]\d$",
        description="免打扰结束时间(HH:mm:ss)",
    )
    preferences: dict[str, Any] | None = None


class AnnouncementForm(BaseModel):
    title: str = Field(..., min_length=2, max_length=50, description="公告标题(2-50字符)")
    content: str = Field(..., min_length=1, description="公告内容")
    type: str = Field(..., min_length=1, description="公告类型")
    importance: int = Field(..., ge=1, le=2, description="重要级别(1:普通;2:重要)")
    targetScope: str = Field(..., min_length=1, description="发送范围")
    targetParams: dict[str, Any] | None = None
    sendTime: str | None = None
    expireTime: str | None = None


class AnnouncementUpdateForm(BaseModel):
    title: str | None = Field(
        default=None, min_length=2, max_length=50, description="公告标题(2-50字符)"
    )
    content: str | None = Field(default=None, min_length=1, description="公告内容")
    type: str | None = Field(default=None, min_length=1, description="公告类型")
    importance: int | None = Field(default=None, ge=1, le=2, description="重要级别(1:普通;2:重要)")
    targetScope: str | None = Field(default=None, min_length=1, description="发送范围")
    targetParams: dict[str, Any] | None = None
    sendTime: str | None = None
    expireTime: str | None = None


class MessageTemplateForm(BaseModel):
    name: str | None = None
    titleTemplate: str | None = None
    contentTemplate: str | None = None
    priority: int | None = Field(default=None, ge=1, le=4)
    channels: dict[str, bool] | None = None
    status: int | None = Field(default=None, ge=0, le=1)

File: dehaze-go/internal/model/bo/message.go
package bo

import "encoding/json"

// MessageSendForm 内部消息发送表单
type MessageSendForm struct {
	TemplateCode string                 `json:"templateCode"`
	Type         string                 `json:"type"`
	Title        string                 `json:"title"`
	Content      string                 `json:"content"`
	RecipientIDs []int64                `json:"recipientIds"`
	BizModule    string                 `json:"bizModule"`
	BizID        string                 `json:"bizId"`
	Priority     int                    `json:"priority"`
	JumpURL      string                 `json:"jumpUrl"`
	Variables    map[string]string      `json:"variables"`
	Extra        map[string]interface{} `json:"extra"`
}

// AnnouncementForm 公告表单
type AnnouncementForm struct {
	Title        string                 `json:"title"`
	Content      string                 `json:"content"`
	Type         string                 `json:"type"`
	Importance   int                    `json:"importance"`
	TargetScope  string                 `json:"targetScope"`
	TargetParams map[string]interface{} `json:"targetParams"`
	SendTime     *string                `json:"sendTime"`
	ExpireTime   *string                `json:"expireTime"`
}

// MessageTemplateForm 模板编辑表单
type MessageTemplateForm struct {
	Name            *string                `json:"name"`
	TitleTemplate   *string                `json:"titleTemplate"`
	ContentTemplate *string                `json:"contentTemplate"`
	Priority        *int                   `json:"priority"`
	Channels        map[string]bool        `json:"channels"`
	Status          *int                   `json:"status"`
}

// NotificationSettingForm 通知设置表单
type NotificationSettingForm struct {
	PushEnabled *bool                  `json:"pushEnabled"`
	DndEnabled  *bool                  `json:"dndEnabled"`
	DndStart    *string                `json:"dndStart"`
	DndEnd      *string                `json:"dndEnd"`
	Preferences *NotificationPrefsForm `json:"preferences"`
}

// NotificationPrefsForm 通知偏好子表单
type NotificationPrefsForm struct {
	TypeChannels   map[string]TypeChannelForm `json:"typeChannels"`
	ModuleSwitches map[string]bool            `json:"moduleSwitches"`
}

// TypeChannelForm 类型渠道开关表单
type TypeChannelForm struct {
	Push bool `json:"push"`
}

// ToJSONString 将map转为JSON字符串
func ToJSONString(v any) string {
	if v == nil {
		return ""
	}
	b, err := json.Marshal(v)
	if err != nil {
		return ""
	}
	return string(b)
}
